libobs/obs-hotkey.c

```diff
@@ -1021,59 +905,55 @@ static void release_registerer(obs_hotkey_t *hotkey)
 	hotkey->registerer = NULL;
 }
 
-static inline bool unregister_hotkey(obs_hotkey_id id)
+static inline void unregister_hotkey(obs_hotkey_id id)
 {
 	if (id >= obs->hotkeys.next_id)
-		return false;
-
-	size_t idx;
-	if (!find_id(id, &idx))
-		return false;
+		return;
 
-	obs_hotkey_t *hotkey = &obs->hotkeys.hotkeys.array[idx];
+	obs_hotkey_t *hotkey;
+	HASH_FIND_HKEY(obs->hotkeys.hotkeys, id, hotkey);
+	if (!hotkey)
+		return;
 
 	hotkey_signal("hotkey_unregister", hotkey);
 
 	release_registerer(hotkey);
 
+	if (hotkey->registerer_type == OBS_HOTKEY_REGISTERER_SOURCE)
+		obs_weak_source_release(hotkey->registerer);
+
 	bfree(hotkey->name);
 	bfree(hotkey->description);
+	bfree(hotkey);
 
-	if (hotkey->registerer_type == OBS_HOTKEY_REGISTERER_SOURCE)
-		obs_weak_source_release(hotkey->registerer);
+	HASH_DEL(obs->hotkeys.hotkeys, hotkey);
```

Comment:
Should `HASH_DEL` be called before `bfree`?
```suggestion
	HASH_DEL(obs->hotkeys.hotkeys, hotkey);

	bfree(hotkey);
```

Reply:
Oh yes, this is a use after free so they must be the other way around.

Reply:
With this fix, the failures I reported at https://github.com/obsproject/obs-studio/pull/8229#issuecomment-1473194490 never happen again :-)